normalizer/src/main/scala/tech/sourced/berserker/normalizer/Main.scala

```diff
@@ -39,13 +43,40 @@ object Main extends App {
     .getOrCreate()
 
   // Start gRPC connection
-  val extractorService = ExtractorService(grpcHost, grpcPort, grpcPlainText)
 
-  val dataRDD = spark.sparkContext.parallelize(extractorService.getRepositoriesData)
+  val extractorService = ExtractorService(grpcHost, grpcPort, grpcMaxMsgSize, grpcPlainText)
+
+  val dataRDD = queryMetadataDbForAllFetchRepos()
+    .repartition(numberOfWorkers)
+    .mapPartitions(partition => {
+      //TODO(bzz): start executor-server Golang binary
+      extractorService.getRepositoriesData(partition.toSeq).toIterator
+    })
+
   val dataDF = spark.sqlContext.createDataFrame(dataRDD, Schema.files)
 
   dataDF.write
     .mode(parquetMode)
     .parquet(parquetFilename)
   dataDF.show(10)
+
+
+  def queryMetadataDbForAllFetchRepos(n: Int = 0): RDD[String] = {
+    import spark.implicits._
+    val jdbcDF = spark.read
+      .format("jdbc")
+      .option("url", "jdbc:postgresql://localhost:5432/testing")
+      .option("dbtable", "repositories")
+      .option("user", "testing")
+      .option("password", "testing")
+      .load()
+
+    val ids = if (n > 0) {
+      jdbcDF.select($"id").limit(n)
+    } else {
+      jdbcDF.select($"id")
+    }
+    return ids.rdd.map(_.getString(0))
```

Comment:
why do you need to return an RDD instead of a DataFrame? we can avoid this map step here.